pymc_marketing/clv/models/modified_beta_geo.py

```diff
@@ -0,0 +1,316 @@
+
+        Parameters
+        ----------
+        future_t : int, array_like
+            Number of time periods to predict expected purchases.
+        data : ~pandas.DataFrame
+            Optional dataframe containing the following columns:
+
+            * `customer_id`: Unique customer identifier
+            * `frequency`: Number of repeat purchases
+            * `recency`: Time between the first and the last purchase
+            * `T`: Time between first purchase and end of observation period; model assumptions require T >= recency
+
+        References
+        ----------
+        .. [1] Fader, Peter S., Bruce G.S. Hardie, and Ka Lok Lee (2005a),
+            "Counting Your Customers the Easy Way: An Alternative to the
+            Pareto/NBD Model," Marketing Science, 24 (2), 275-84.
+            https://www.brucehardie.com/papers/bgnbd_2004-04-20.pdf
+
+        """  # noqa: E501
+        if data is None:
+            data = self.data
+
+        if future_t is not None:
+            data = data.assign(future_t=future_t)
+
+        dataset = self._extract_predictive_variables(
+            data, customer_varnames=["frequency", "recency", "T", "future_t"]
+        )
+        a = dataset["a"]
+        b = dataset["b"]
+        alpha = dataset["alpha"]
+        r = dataset["r"]
+        x = dataset["frequency"]
+        t_x = dataset["recency"]
+        T = dataset["T"]
+        t = dataset["future_t"]
+
+        hyp_term = hyp2f1(r + x, b + x + 1, a + b + x, t / (alpha + T + t))
+        first_term = (a + b + x) / (a - 1)
+        second_term = 1 - hyp_term * ((alpha + T) / (alpha + t + T)) ** (r + x)
+        numerator = first_term * second_term
+        denominator = 1 + (a / (b + x)) * ((alpha + T) / (alpha + t_x)) ** (r + x)
+
+        return (numerator / denominator).transpose(
+            "chain", "draw", "customer_id", missing_dims="ignore"
+        )
+
+    def expected_purchases_new_customer(
+        self,
+        data: pd.DataFrame | None = None,
+        *,
+        t: int | np.ndarray | pd.Series | None = None,
+    ) -> xarray.DataArray:
+        r"""Compute the expected number of purchases for a new customer across *t* time periods.
+
+        Adapted from equation (9) in [1]_, and `lifetimes` library:
+        https://github.com/CamDavidsonPilon/lifetimes/blob/41e394923ad72b17b5da93e88cfabab43f51abe2/lifetimes/fitters/beta_geo_fitter.py#L328
+
+        Parameters
+        ----------
+        t : array_like
+            Number of time periods over which to estimate purchases.
+
+        References
+        ----------
+        .. [1] Fader, Peter S., Bruce G.S. Hardie, and Ka Lok Lee (2005a),
+            "Counting Your Customers the Easy Way: An Alternative to the
+            Pareto/NBD Model," Marketing Science, 24 (2), 275-84.
+            http://www.brucehardie.com/notes/021/palive_for_BGNBD.pdf
+
+        """
```

Comment:
These docstrings need to be updated to reflect the current model.

Reply:
Updated